test/ViewExtensionLibraryTests/LibraryResourceProviderTests.cs

```diff
@@ -290,13 +293,212 @@ public void PackagedCustomNodeSearchElementLoadedType()
             var moq = new Mock<ICustomNodeSource>();
             var element = new CustomNodeSearchElement(moq.Object, info);
 
-            var provider = new NodeItemDataProvider(null);
+            var provider = new NodeItemDataProvider(new NodeSearchModel());
             var item = provider.CreateLoadedTypeItem(element);
             Assert.AreEqual(expectedQualifiedName, item.fullyQualifiedName);
             Assert.AreEqual(guid.ToString(), item.contextData);
             Assert.AreEqual(string.Empty, item.keywords);
             var url = new IconUrl(name, path, true);
             Assert.AreEqual(url.Url, item.iconUrl);
         }
+
+        [Test]
+        [Category("UnitTests")]
+        public void LibraryDataUpdatedEventRaised()
+        {
+            const string libraryDataUpdated = "libraryDataUpdated";
+            var timeout = 50; //50 milliseconds
+            var resetevent = new AutoResetEvent(false);
+
+            var model = new NodeSearchModel();
+            var controller = new Mock<IEventController>();
+            controller.Setup(c => c.RaiseEvent(It.IsAny<string>(), It.IsAny<object[]>())).Callback(() => resetevent.Set());
```

Comment:
Mock objects don't execute in the background thread, but the RaiseEvent method will be called on TimerCallback and that will run in a background thread.